collect_app/src/main/java/org/odk/collect/android/tasks/DownloadFormListTask.java

```diff
@@ -144,6 +145,7 @@ private boolean isXformsListNamespacedElement(Element e) {
                 String description = null;
                 String downloadUrl = null;
                 String manifestUrl = null;
+                String submissionUrl = null;
```

Comment:
Forms often have empty submission URLs. If this is set to null, you won't be able to distinguish between an empty submission URL (which is fine) and a bad submission URL. You'll hit line 218 every time you try to parse a blank form before download. You can reproduce this by checking out this branch and trying to download new forms.